src/iuse.cpp

```diff
@@ -2012,41 +2012,79 @@ int iuse::sew_advanced(player *p, item *it, bool, const tripoint& )
 
     // TODO: Wrap all the mods into structs, maybe even json-able
     // All possible mods here
-    std::array<std::string, 4> clothing_mods{
+    const std::array<const std::string, 4> clothing_mods{
         { "wooled", "furred", "leather_padded", "kevlar_padded" }
     };
 
     // Materials those mods use
-    std::array<std::string, 4> mod_materials{
+	const std::array<const std::string, 4> mod_materials{
         { "felt_patch", "fur", "leather", "kevlar_plate" }
     };
 
+    // TODO: Read data from \data\json\items\generic\filament.json
+    // Filament that can be used
+	const std::array<const std::string, 4> mod_filament{
+		{ "thread", "sinew", "plant_fibre", "yarn" }
+	};
```

Comment:
A better way than hardcode would be to create a `requirement` for filament and check that. That way it would not be hardcoded and you would not need to manually replicate what crafting process already does.
You access requirement by `requirement_id( "id_here" ).obj()`, and use it like:
```
const auto &req = making->requirements();
bool can_craft = req.can_make_with_inventory( crafting_inv, batch_size );
```

Then consume the resources:
```
for( const auto &it : req.get_components() ) {
    consume_items( it, batch_size );
}
for( const auto &it : req.get_tools() ) {
    consume_tools( it, batch_size );
}
```

Doing it this way would automatically ask for resources, so you may skip the second part and just consume manually. Still, the first part is great for asking if resources exist.